activation/activation.go

```diff
@@ -88,8 +87,8 @@ type Config struct {
 // it is responsible for initializing post, receiving poet proof and orchestrating nipst. after which it will
 // calculate total weight and providing relevant view as proof.
 type Builder struct {
-	pendingPoetClient atomic.UnsafePointer
-	started           atomic.Bool
+	pendingPoetClients chan []PoetProvingServiceClient
```

Comment:
The code becomes easier to understand if you use a `sync.Mutex` to protect the field of the struct instead of a channel.

```suggestion
	pendingPoetClientsMux sync.Mutex
	pendingPoetClients    []PoetProvingServiceClient
```

Reply:
Refactored